common/authorization/oauthAuthorizer.go

```diff
@@ -107,19 +127,40 @@ func (a *oauthAuthority) parseToken(tokenStr string, verifier jwt.Verifier) (*jw
 	return &claims, nil
 }
 
-func (a *oauthAuthority) validateClaims(claims *jwtClaims, attributes *Attributes) error {
+func (a *oauthAuthority) validateTTL(claims *jwtClaims) error {
 	if claims.TTL > a.authorizationCfg.MaxJwtTTL {
 		return fmt.Errorf("TTL in token is larger than MaxTTL allowed")
 	}
 	if claims.Iat+claims.TTL < time.Now().Unix() {
 		return fmt.Errorf("JWT has expired")
 	}
-	if claims.Domain != attributes.DomainName {
-		return fmt.Errorf("domain in token doesn't match with current domain")
-	}
-	if NewPermission(claims.Permission) < attributes.Permission {
-		return fmt.Errorf("token doesn't have the right permission")
+	return nil
+}
+
+func (a *oauthAuthority) validatePermission(claims *jwtClaims, attributes *Attributes, data map[string]string) error {
+	groups := ""
+	switch attributes.Permission {
+	case PermissionRead:
+		groups = data[common.DomainDataKeyForReadGroups] + groupSeparator + data[common.DomainDataKeyForWriteGroups]
+	case PermissionWrite:
+		groups = data[common.DomainDataKeyForWriteGroups]
+	default:
+		if claims.Admin {
```

Comment:
This is test in line 91 so maybe we should only leave 

`return fmt.Errorf("token doesn't have permission for %s API", attributes.Permission)` (in case we add more in the future)

Reply:
yeah I am fine with that.